docs/design/mp_overview.md

```diff
@@ -0,0 +1,186 @@
+number of servers to prevent failures to cascade between components.
+Also having "One" daemon-set exposes multiple extensible RPC
+interfaces minimizes the number of agents on the hosts, improves
+deployment story, and reduces the chances of cycle-stealing from
+actual workload on the host.
+
+We split Objects and Operators into two main categories: Generic, and
+DataPlane specific. The Generic objects are necessary for any
+cloud-networking solutions such as: VPC (or Network in Neutron
+terminology), Network (or Subnet in Neutron), Endpoint (or Port in
+Neutron), And Droplet which specifies a host (Or a Worker node in
+Kubernetes). The other category of objects are those that are
+data-plane specific; For Mizar these are Bouncers and Dividers
+Objects.  Each Object's must specify a "status" field that indicates
+the status of the object in the pipeline. Two statuses are necessary:
+Init, and Provisioned. In the PoC, the Management plane has four
+generic objects: Droplet, VPC, Network, and Endpoint. Each Object is
+specified using Custom Resource Definitions (CRDs).
+
+Each Operator is responsible for maintaining the data of one and only
+one object (Stateful Object). We call this a Stateful Object from the
+operator perspective, since the Operator maintains its data. It is
+mandatory that an Operator updates the data for its Object when it is
+in the Provisioned state and optional in any other state. Operators
+may also mutate the state of objects other than its Stateful Object,
+but never store these objects data. We call these Objects Mutable
+objects from an Operator perspective. The Objects types with respects
+to Operators will become clear as we detail Mizar's workflow.
+
+The specification of an Operator's Stateful Object, Mutable Object,
+and Mutating Actions are what defines the management plane workflow.
+
+### Inter-Operators Communications
+
+Operators communicate with each other to evolve the object's state.
+When an operator mutates the state of the object it invokes the next
+operator(s) in the object pipeline. The invocation is done through a
+callback mechanism. According to the deployment and specific
+implementation the callback is translated into one of multiple modes.
+In all of these modes, the operator developer only focuses on the
+workflow through calling the callback function without worrying about
+the details of the communication implementation.
+
+1. **Through the API Server(s)**: In this mode, operators communication is
+   implicit with the help of the API server. As one operator makes the
+   callback to invoke the next step of the object pipeline, the
+   only changes the object state and update the object rhough the API
+   server. As the next Operator in the pipeline handles the object in
+   the new state, communication happens implicitly. The drawback of
+   this approach is the loading on the API server, the latency
+   involved to evolve the object's pipeline. The advantage is that the
```

Comment:
Here I think we should say:

The drawback of this approach is the **load** on the API server, **and** the latency involved to evolve the object's pipeline.

Reply:
ok